acceptance/setup/aio/pre-suite/010_Install.rb

```diff
@@ -47,19 +56,33 @@
       master.install_package('puppetserver')
 
       logger.info "EC2 master found: Installing #{ENV['SHA']} build of puppet-agent."
-      dev_builds_url = ENV['DEV_BUILDS_URL'] || 'http://builds.delivery.puppetlabs.net'
-      install_from_build_data_url('puppet-agent', "#{dev_builds_url}/puppet-agent/#{ENV['SHA']}/artifacts/#{ENV['SHA']}.yaml", master)
+      # Upgrade installed puppet-agent with targeted SHA.
+      if dev_builds_url == 'http://builds.delivery.puppetlabs.net'
+        base_url, build_details = fetch_build_details("#{dev_builds_url}/puppet-agent/#{ENV['SHA']}/artifacts/#{ENV['SHA']}.yaml")
+        artifact_url, _ = host_urls(master, build_details, base_url)
+        release_path = File.dirname(artifact_url)
+        release_file = File.basename(artifact_url)
+      else
+        release_path_end, release_file = master.puppet_agent_dev_package_info( opts[:puppet_collection], opts[:puppet_agent_version], opts)
+        release_path = "#{opts[:dev_builds_url]}/puppet-agent/#{opts[:puppet_agent_sha]}/repos/"
+        release_path << release_path_end
+      end
+      copy_dir_local = File.join('tmp', 'repo_configs', master['platform'])
+      fetch_http_file(release_path, release_file, copy_dir_local)
+      scp_to master, File.join(copy_dir_local, release_file), master.external_copy_base
+      on master, "rpm -Uvh #{File.join(master.external_copy_base, release_file)} --oldpackage --force"
     else
       fail_test("EC2 master found, but it was not an `el` host: The specified `puppet-agent` build (#{ENV['SHA']}) cannot be installed.")
     end
   else
-    dev_builds_url = ENV['DEV_BUILDS_URL'] || "http://builds.delivery.puppetlabs.net"
-    install_from_build_data_url('puppet-agent', "#{dev_builds_url}/puppet-agent/#{ENV['SHA']}/artifacts/#{ENV['SHA']}.yaml", master)
-    if ENV['SERVER_VERSION'].nil? || ENV['SERVER_VERSION'] == 'latest'
-      install_puppetlabs_dev_repo(master, 'puppetserver', 'latest', nil, :dev_builds_url => 'http://nightlies.puppet.com')
-      master.install_package('puppetserver')
-    else
+    if ENV['SERVER_VERSION'] && ENV['SERVER_VERSION'] != 'latest' && dev_builds_url == 'http://builds.delivery.puppetlabs.net'
+      install_from_build_data_url('puppet-agent', "#{dev_builds_url}/puppet-agent/#{ENV['SHA']}/artifacts/#{ENV['SHA']}.yaml", master)
       install_from_build_data_url('puppetserver', "#{dev_builds_url}/puppetserver/#{ENV['SERVER_VERSION']}/artifacts/#{ENV['SERVER_VERSION']}.yaml", master)
+    else
+      server_version = ENV['SERVER_VERSION'] || 'latest'
+      install_puppetlabs_dev_repo(master, 'puppetserver', server_version, nil, :dev_builds_url => dev_builds_url)
```

Comment:
this will fail if `server_version` is `latest` and `dev_builds_url` is `builds.delivery.puppetlabs.net`

Reply:
I think the fix from https://github.com/puppetlabs/pxp-agent/pull/663 can also be applied here.

Reply:
Fix applied, we should be in the clear now